### DISCUSSION TITLE
Question: Use Costom Popover Component with CharacterStylesPopupPlugin

### DISCUSSION THREAD

**Discussion by WinmezzZ** [facebook/lexical#2408] [General]
I want create a inline toolbar, in playground-demo it called `CharacterStylesPopupPlugin`, it set position by native dom operation, but not update when dragging browser window.

How can I use my custom popover component? Like [antd popover](https://ant.design/components/popover/) or [mui popover](https://mui.com/material-ui/react-popover/#main-content), shoud I custom [decorator nodes](https://lexical.dev/docs/concepts/nodes#decoratornode) ?

I would appreciate any demo or ideas provided.

Thank you.

**Comment by trueadm** [comment]
We should probably add a ResizeObserver to ensure it gets positioned when the window changes.

We don't recommend using decorator nodes for popovers, but instead position them using a portal over the top like the CharacterStylesPopup does. This offers the best user experience without affecting the content of state of the editor (a decorator node would be in the state of the editor, and it doesn't make sense for a decorator's concern to be about popups).
